gensim/models/doc2vec.py

```diff
@@ -908,10 +908,7 @@ def infer_vector(self, doc_words, alpha=None, min_alpha=None, epochs=None, steps
 
         """
         if isinstance(doc_words, string_types):
-            warnings.warn(
-                "The parameter `doc_words` takes a list of strings, while you passed a single string, "
-                "which will be accepted and treated as a list of characters. Is this what you wanted?"
-            )
+            raise ValueError("You must pass a list of strings, not a single string.")
```

Comment:
Perhaps this should be a `TypeError` - because all string-types will cause the error. And naming the method or parameter would make the error easier to interpret, without looking further up the stack dump (which occasionally by user preferences might not even be shown.) For example: "Parameter `doc_words` of `infer_vector()` must be a list of strings (not a single string)."